hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java

```diff
@@ -144,13 +147,24 @@ public void handle(@NonNull final HandleContext context) throws HandleException
 
         for (final var xfers : op.tokenTransfers()) {
             final var tokenId = xfers.tokenOrThrow();
-            validateTrue(tokenHasNoCustomFeesPaidByReceiver(tokenId, tokenStore), INVALID_TRANSACTION);
-
+            final var token = getIfUsable(tokenId, tokenStore);
             boolean shouldExecuteCryptoTransfer = false;
             final var transferListBuilder = TokenTransferList.newBuilder().token(tokenId);
 
             // process fungible token transfers if any
             if (!xfers.transfers().isEmpty()) {
+                for (var transfer : xfers.transfers()) {
```

Comment:
Consider moving this entire loop into a method - or the body of the loop (or both).  Would make it possible to _unit test_.  

Same at L222-227 for nfts.

This method `handle` is incredibly complex - far too complex for reasonable testing.  THis code here is a loop in a conditional in a loop.  140 lines in one method ... not a good idea.

Reply:
I agree, but since this method is implemented in https://github.com/hashgraph/hedera-services/pull/14279 - we will refactor it there 👍

Reply:
ah, ok! thank you!